apps/openmw/mwworld/worldimp.cpp

```diff
@@ -1024,9 +1025,17 @@ namespace MWWorld
 
             float activationDistance = getMaxActivationDistance() + telekinesisRangeBonus;
 
-            facedObject = getFacedObject(activationDistance);
+            facedObject = getFacedObject(activationDistance, distanceToObject, true);
+
+            // Not allowing telekinesis on actors, on doors that teleport to other cells, or on activators
+            // Original engine doesn't allow telekinesis on books or lights, either
+            if (!facedObject.isEmpty() && (facedObject.getClass().isActor()
+                || facedObject.getCellRef().getTeleport() || facedObject.getClass().getTypeName() == typeid(ESM::Activator).name())
+                && (distanceToObject > getMaxActivationDistance()))
+                return 0;
         }
 
+        mDistanceToFacedObject = distanceToObject;
```

Comment:
If no object was hit, an uninitialized value is assigned to mDistanceToFacedObject.